### PR TITLE
Siyao h/1.17/gestures

### PR DESCRIPTION
**Pull request by SiyaoHuang**

### CODE REVIEW

File: com.unity.uiwidgets/Runtime/gestures/converter.cs [hunk 1]
```diff
@@ -4,300 +4,213 @@
...
-                        if (state == null || !state.down) {
                             break;
-                        }
 
-                        D.assert(state.down);
-                        if (position != state.lastPosition) {
-                            Offset offset = position - state.lastPosition;
-                            state.lastPosition = position;
+                        case PointerChange.move: {
+                            Debug.LogError("need add synthesized into PointerData");
```
Comment (zhuxingwei) [inline com.unity.uiwidgets/Runtime/gestures/converter.cs:106]: why add this Debug.LogError line?

File: com.unity.uiwidgets/Runtime/gestures/multitap.cs [hunk 2]
```diff
@@ -228,6 +258,16 @@ public DoubleTapGestureRecognizer(object debugOwner = null, PointerDeviceKind? k
             }
         }
 
+        void _checkUp(_TapTracker tracker) {
+            // D.assert(buttons == kPrimaryButton);
```
Comment (zhuxingwei) [inline com.unity.uiwidgets/Runtime/gestures/multitap.cs:262]: why comment this assert out?

File: com.unity.uiwidgets/Runtime/gestures/multitap.cs [hunk 3]
```diff
@@ -228,6 +258,16 @@ public DoubleTapGestureRecognizer(object debugOwner = null, PointerDeviceKind? k
...
+            if (onDoubleTap != null) {
+                invokeCallback<object>("onDoubleTap", () => {
+                    onDoubleTap(new DoubleTapDetails(tracker._initialGlobalPosition));
```
Comment (zhuxingwei) [inline com.unity.uiwidgets/Runtime/gestures/multitap.cs:265]: the delegate GestureDoubleTapCallback should take zero argument, is it a typo here?

File: com.unity.uiwidgets/Runtime/gestures/converter.cs (hunk 1 above)
Comment (SiyaoHuang) [reply com.unity.uiwidgets/Runtime/gestures/converter.cs:106]: This is for updating master. I forgot to remove it.

File: com.unity.uiwidgets/Runtime/gestures/multitap.cs (hunk 2 above)
Comment (SiyaoHuang) [reply com.unity.uiwidgets/Runtime/gestures/multitap.cs:262]: fixed

File: com.unity.uiwidgets/Runtime/gestures/multitap.cs (hunk 3 above)
Comment (SiyaoHuang) [reply com.unity.uiwidgets/Runtime/gestures/multitap.cs:265]: fixed

**Pull request merged**